### PR TITLE
update to sdk version 0.17.1

### PR DESCRIPTION
**Pull request by d-shapiro**

### CODE REVIEW

File: src/main/java/bacnet/BacnetPoint.java [hunk 1]
```diff
@@ -282,10 +280,10 @@ private void setupNode() {
...
-		if (node.getChild("present value") == null) {
-			node.createChild("present value").setValueType(ValueType.STRING).setValue(new Value("")).build();
+		if (node.getChild("present value", true) == null) {
+			node.createChild("present value", true).setValueType(ValueType.STRING).setValue(new Value("")).build();
 		}
-		node.getChild("present value").setWritable(Writable.NEVER);
+		node.getChild("present value", true).setWritable(Writable.NEVER);
 		folder.conn.link.setupPoint(this, folder);
 
 		// if (DeviceFolder.isOneOf(oid.getObjectType(), ObjectType.trendLog)) {
```
Comment (Pacane) [inline src/main/java/bacnet/BacnetPoint.java]: remove if unused.

File: src/main/java/bacnet/DeviceFolder.java [hunk 2]
```diff
@@ -550,9 +550,9 @@ protected void duplicate(String name) {
 		JsonObject jobj = conn.link.copySerializer.serialize();
 		JsonObject parentobj = getParentJson(jobj, node);
 		JsonObject nodeobj = parentobj.get(node.getName());
-		parentobj.put(name, nodeobj);
+		parentobj.put(org.dsa.iot.dslink.util.StringUtils.encodeName(name), nodeobj);
```
Comment (Pacane) [inline src/main/java/bacnet/DeviceFolder.java:553]: Any reason for the full name here?

File: src/main/java/bacnet/LocalDeviceNode.java [hunk 3]
```diff
@@ -92,7 +92,7 @@ public void handle(ActionResult event) {
 					// enable(true);
```
Comment (Pacane) [inline src/main/java/bacnet/LocalDeviceNode.java]: remove if unused

**Review by Pacane — CHANGES_REQUESTED** [review]

File: src/main/java/bacnet/DeviceFolder.java (hunk 2 above)
Comment (d-shapiro) [reply src/main/java/bacnet/DeviceFolder.java:553]: org.apache.commons.lang3.StringUtils

**Review by hhou2011 — APPROVED** [review]

**Pull request merged**